cmd/ndm_daemonset/probe/usedbyprobe.go

```diff
@@ -0,0 +1,211 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"errors"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/openebs/node-disk-manager/blockdevice"
+	"github.com/openebs/node-disk-manager/cmd/ndm_daemonset/controller"
+	"github.com/openebs/node-disk-manager/pkg/blkid"
+	"github.com/openebs/node-disk-manager/pkg/spdk"
+	libudevwrapper "github.com/openebs/node-disk-manager/pkg/udev"
+	"github.com/openebs/node-disk-manager/pkg/util"
+
+	"k8s.io/klog"
+)
+
+type usedbyProbe struct {
+	Controller      *controller.Controller
+	BlkidIdentifier *blkid.DeviceIdentifier
+}
+
+const (
+	usedbyProbeConfigKey = "used-by-probe"
+	usedbyProbePriority  = 4
+
+	k8sLocalVolumePath1 = "kubernetes.io/local-volume"
+	k8sLocalVolumePath2 = "kubernetes.io~local-volume"
+	zfsFileSystemLabel  = "zfs_member"
+)
+
+var (
+	usedbyProbeName  = "used-by probe"
+	usedbyProbeState = defaultEnabled
+)
+
+var usedbyProbeRegister = func() {
+	// Get a controller object
+	ctrl := <-controller.ControllerBroadcastChannel
+	if ctrl == nil {
+		klog.Error("unable to configure", usedbyProbeName)
+		return
+	}
+	if ctrl.NDMConfig != nil {
+		for _, probeConfig := range ctrl.NDMConfig.ProbeConfigs {
+			if probeConfig.Key == usedbyProbeConfigKey {
+				usedbyProbeName = probeConfig.Name
+				usedbyProbeState = util.CheckTruthy(probeConfig.State)
+				break
+			}
+		}
+	}
+	newRegisterProbe := &registerProbe{
+		priority:   usedbyProbePriority,
+		name:       usedbyProbeName,
+		state:      usedbyProbeState,
+		pi:         &usedbyProbe{Controller: ctrl},
+		controller: ctrl,
+	}
+	// Here we register the used-by probe
+	newRegisterProbe.register()
+}
+
+func newUsedByProbe(devPath string) *usedbyProbe {
+	usedbyProbe := &usedbyProbe{
+		BlkidIdentifier: &blkid.DeviceIdentifier{
+			DevPath: devPath,
+		},
+	}
+	return usedbyProbe
+}
+
+func (sp *usedbyProbe) Start() {}
+
+func (sp *usedbyProbe) FillBlockDeviceDetails(blockDevice *blockdevice.BlockDevice) {
+	if blockDevice.DevPath == "" {
+		klog.Errorf("device identifier found empty, used-by probe will not fetch information")
+		return
+	}
+
+	// checking for local PV on the device
+	for _, mountPoint := range blockDevice.FSInfo.MountPoint {
+		if strings.Contains(mountPoint, k8sLocalVolumePath1) ||
+			strings.Contains(mountPoint, k8sLocalVolumePath2) {
+			blockDevice.DevUse.InUse = true
+			blockDevice.DevUse.UsedBy = blockdevice.LocalPV
+			klog.V(4).Infof("device: %s Used by: %s filled by used-by probe", blockDevice.DevPath, blockDevice.DevUse.UsedBy)
+			return
+		}
+	}
+
+	// checking for cstor and zfs localPV
+	// we start with the assumption that device has a zfs file system
+	hasZFS := true
```

Comment:
hasZFS is little confusing, can we make it  lookupZFS something like this?

Reply:
Done.